src/java.base/share/classes/java/lang/ProcessHandleImpl.java

```diff
@@ -167,6 +169,8 @@ public void run() {
                         newCompletion.complete(exitValue);
                         // remove from cache afterwards
                         completions.remove(pid, newCompletion);
+                        // Restore thread name
```

Comment:
Just wondering if we the name should be restored in a finally block in case there are scenarios where the task might throw an exception. It might not be necessary, I would need to page in the details on the  thread pool configuration to see if the thread terminates when a task fails with an exception.

Reply:
This makes sense, thanks for the suggestion. I'll change the patch.

Reply:
The updated patch looks okay but style-wise it means more indentation and the code gets pushed to the right which is a bit annoying for side-by-side review.  @RogerRiggs is the main maintainer in this area and may wish to look at this change, maybe the task body could be moved to a separate method so that its not inlined in the completion method.